src/jit/codegenxarch.cpp

```diff
@@ -7570,18 +7570,42 @@ void CodeGen::genCompareInt(GenTreePtr treeNode)
 {
     assert(treeNode->OperIsCompare());
 
-    GenTreeOp* tree    = treeNode->AsOp();
-    GenTreePtr op1     = tree->gtOp1;
-    GenTreePtr op2     = tree->gtOp2;
-    var_types  op1Type = op1->TypeGet();
-    var_types  op2Type = op2->TypeGet();
+    GenTreeOp* tree      = treeNode->AsOp();
+    GenTreePtr op1       = tree->gtOp1;
+    GenTreePtr op2       = tree->gtOp2;
+    var_types  op1Type   = op1->TypeGet();
+    var_types  op2Type   = op2->TypeGet();
+    regNumber  targetReg = treeNode->gtRegNum;
+
+#ifdef FEATURE_SIMD
+    // If we have GT_JTRUE(GT_EQ/NE(GT_SIMD((in)Equality, v1, v2), true/false)),
+    // then we don't need to generate code for GT_EQ/GT_NE, since SIMD (in)Equality intrinsic
+    // would set or clear Zero flag.
```

Comment:
This is just a little confusing, because this is a case where both `treeNode` and `op1` do not require a register. It might be worth breaking out the example, e.g.
simdCompareResult = GT_SIMD((In)Equality, v1, v2)
integerCompareResult = GT_EQ/NE(simdCompareResult, true/false)
GT_JTRUE(integerCompareResult)
And mention that for this case we don't need to generate either CompareResult into a register.